src/Auctions/Auction.sol

```diff
@@ -0,0 +1,680 @@
+        uint256 hoursComponent = 1e27 >> (secondsElapsed / 3600);
+        uint256 minutesComponent = Maths.rpow(
+            MINUTE_HALF_LIFE,
+            (secondsElapsed % 3600) / 60
+        );
+        uint256 initialPrice = Maths.wdiv(startingPrice * 1e18, _available);
+
+        return
+            (initialPrice * Maths.rmul(hoursComponent, minutesComponent)) /
+            1e27;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            SETTERS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Enables a new auction.
+     * @dev Uses governance as the receiver.
+     * @param _from The address of the token to be auctioned.
+     * @return . The unique identifier of the enabled auction.
+     */
+    function enable(address _from) external virtual returns (bytes32) {
+        return enable(_from, msg.sender);
+    }
+
+    /**
+     * @notice Enables a new auction.
+     * @param _from The address of the token to be auctioned.
+     * @param _receiver The address that will receive the funds in the auction.
+     * @return _auctionId The unique identifier of the enabled auction.
+     */
+    function enable(
+        address _from,
+        address _receiver
+    ) public virtual onlyGovernance returns (bytes32 _auctionId) {
+        address _want = want();
+        require(_from != address(0) && _from != _want, "ZERO ADDRESS");
+        require(
+            _receiver != address(0) && _receiver != address(this),
+            "receiver"
+        );
+
+        // Calculate the id.
+        _auctionId = getAuctionId(_from);
+
+        require(
+            auctions[_auctionId].fromInfo.tokenAddress == address(0),
+            "already enabled"
+        );
+
+        // Store all needed info.
+        auctions[_auctionId].fromInfo = TokenInfo({
+            tokenAddress: _from,
+            scaler: uint96(WAD / 10 ** ERC20(_from).decimals())
+        });
+        auctions[_auctionId].receiver = _receiver;
+
+        // Add to the array.
+        enabledAuctions.push(_auctionId);
+
+        emit AuctionEnabled(_auctionId, _from, _want, address(this));
+    }
+
+    /**
+     * @notice Disables an existing auction.
+     * @dev Only callable by governance.
+     * @param _from The address of the token being sold.
+     */
+    function disable(address _from) external virtual onlyGovernance {
+        bytes32 _auctionId = getAuctionId(_from);
+
+        // Make sure the auction was enabled.
+        require(
+            auctions[_auctionId].fromInfo.tokenAddress != address(0),
+            "not enabled"
+        );
+
+        // Remove the struct.
+        delete auctions[_auctionId];
+
+        // Remove the auction ID from the array.
+        bytes32[] memory _enabledAuctions = enabledAuctions;
+        for (uint256 i = 0; i < _enabledAuctions.length; ++i) {
```

Comment:
This could potentially be expensive. You could add the index `_i` as a parameter to the function and simply do a check if `_enabledAuctions[_i] == _auctionId`. Its very easy to find the right value off-chain